Clicker/ViewControllers/PollsViewController.swift

```diff
@@ -332,29 +332,44 @@ class PollsViewController: UIViewController {
         newGroupButton.isUserInteractionEnabled = true
     }
     
-    func joinSessionWithCode(with code: String) -> Future<Response<Session>> {
-        return networking(Endpoint.joinSessionWithCode(with: code)).decode()
+    func joinSessionWithCode(with code: String) -> Future<Data> {
+        return networking(Endpoint.joinSessionWithCode(with: code))
     }
     
-    func getSortedPolls(with id: Int) -> Future<Response<[PollsDateModel]>> {
-        return networking(Endpoint.getSortedPolls(with: id)).decode()
+    func getSortedPolls(with id: Int) -> Future<Data> {
+        return networking(Endpoint.getSortedPolls(with: id))
     }
     
     @objc func joinSession() {
         guard let code = codeTextField.text, code != "" else { return }
-        joinSessionWithCode(with: code).chained { sessionResponse -> Future<Response<[PollsDateModel]>> in
-            self.session = sessionResponse.data
-            return self.getSortedPolls(with: sessionResponse.data.id) 
+        joinSessionWithCode(with: code).chained { sessionData -> Future<Data> in
+            let jsonDecoder = JSONDecoder()
+            jsonDecoder.dateDecodingStrategy = .secondsSince1970
+            if let sessionResponse = try? jsonDecoder.decode(Response<Session>.self, from: sessionData) {
+                self.session = sessionResponse.data
+                return self.getSortedPolls(with: sessionResponse.data.id)
+            }
+            // Fail out of get sorted polls call
+            return self.getSortedPolls(with: -1)
         }.observe { [weak self] result in
-            guard let `self` = self, let session = self.session else { return }
-            DispatchQueue.main.async {
-                switch result {
-                case .value(let pollsResponse):
-                    
+            guard let `self` = self else { return }
+            let jsonDecoder = JSONDecoder()
```

Comment:
Lol, I'm not entirely sure about this one but in Socket, I think jsonDecoder became a defined variable at the top - which I think we can also do here to remove duplicate code? I might be wrong though.

```
var jsonDecoder: JSONDecoder = {
        let decoder = JSONDecoder()
        decoder.dateDecodingStrategy = .secondsSince1970
        return decoder
    }()
```

Reply:
ooo great idea